### PR TITLE
Shared alb support

### PR DESCRIPTION
**Pull request by tahir-duplo**
- Wait for target groups to be created implementation while ECS service creation.
- Flag based delete implementation for ECS task definition
- Terraform Resource - Add shared load balancer support for ECS.

### CODE REVIEW

File: duplocloud/resource_duplo_ecs_service.go [hunk 1]
```diff
@@ -55,11 +56,25 @@ func ecsServiceSchema() map[string]*schema.Schema {
...
 		"dns_prfx": {
 			Description: "The DNS prefix to assign to this service's load balancer.",
 			Type:        schema.TypeString,
 			Optional:    true,
 		},
+		"wait_until_targets_ready": {
+			Description:      "Whether or not to wait until all target groups are created for ecs service, after creation.",
+			Type:             schema.TypeBool,
+			Optional:         true,
+			ForceNew:         true,
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_ecs_service.go:74]: we should not force a new resource when this flag changes, right?

File: duplocloud/resource_duplo_ecs_service.go [hunk 2]
```diff
@@ -55,11 +56,25 @@ func ecsServiceSchema() map[string]*schema.Schema {
 			Optional: true,
 			Default:  false,
 		},
+		"target_group_arns": {
+			Type:     schema.TypeSet,
+			Optional: true,
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_ecs_service.go:61]: is the user even allowed to pass these in?

if not, then only `Computed` should be `true`, and `Optional` should be left as `false`.

File: duplocloud/resource_duplo_ecs_task_definition.go [hunk 3]
```diff
@@ -48,6 +48,13 @@ func ecsTaskDefinitionSchema() map[string]*schema.Schema {
 			Type:        schema.TypeString,
 			Computed:    true,
 		},
+		"prevent_tf_destroy": {
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_ecs_task_definition.go:51]: why should we force the resource to be recreated if this flag changes?

File: duplocloud/resource_duplo_tenant.go [hunk 4]
```diff
@@ -177,7 +177,7 @@ func resourceTenantCreate(ctx context.Context, d *schema.ResourceData, m interfa
 	// Wait for 3 minutes to allow tenant creation.
 	if d.Get("wait_until_created").(bool) {
 		log.Printf("[TRACE] resourceTenantCreate(%s): waiting for 2 minutes because 'wait_until_created' is 'true'", rq.AccountName)
-		time.Sleep(time.Duration(2) * time.Minute)
+		time.Sleep(time.Duration(45) * time.Second)
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_tenant.go:180]: ha - thanks.  I've been meaning to do something like this. :)

File: duplosdk/ecs_task_def.go [hunk 5]
```diff
@@ -85,3 +86,44 @@ func (c *Client) EcsTaskDefinitionGet(tenantID, arn string) (*DuploEcsTaskDef, C
 	rp.TenantID = tenantID
 	return &rp, err
 }
+
+// EcsTaskDefinitionDelete deletes an ECS task definition via the Duplo API.
+func (c *Client) EcsTaskDefinitionDelete(tenantID, arn string, preventDestroy bool) ClientError {
```
Comment (joek-duplo) [inline duplosdk/ecs_task_def.go:91]: we should not have TF concepts in here like `preventDestroy`
move that out into the TF provider code, and out of the SDK code.

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: duplocloud/resource_duplo_ecs_service.go (hunk 2 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_ecs_service.go:61]: Done

File: duplosdk/ecs_task_def.go (hunk 5 above)
Comment (tahir-duplo) [reply duplosdk/ecs_task_def.go:91]: Done

File: duplocloud/resource_duplo_ecs_service.go (hunk 2 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_ecs_service.go:61]: I have removed `optional`, which is not required here.

File: duplocloud/resource_duplo_ecs_task_definition.go (hunk 3 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_ecs_task_definition.go:51]: done,

I was facing the following issue without force new, 
Error: InternalValidate
```
Internal validation of the provider failed! This is always a bug
with the provider itself, and not a user issue. Please report
this bug:

1 error occurred:
        * resource duplocloud_ecs_task_definition: No Update defined, must set
ForceNew on: []string{"prevent_tf_destroy"}
```

I have added an updated context empty implementation to resolve this issue.

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**